File: 백준/Silver/2606. 바이러스/바이러스.py
n = int(input())
pair = int(input())
DFS = {i : [] for i in range(1,n + 1)}
for i in range(1,pair + 1):
    a,b = map(int,input().split())
    DFS[a].append(b)
    DFS[b].append(a)
visited = { i : False for i in range(1,n + 1)}
cnt = 0
def f(x):
    global cnt
    for node in DFS[x]:
        if not visited[node]:
            visited[node] = True
            cnt += 1
            f(node)
f(1)
if n == 1:
    print(0)
else:
    print(cnt - 1)

File: 백준/Silver/2606. 바이러스/바이러스.c
#include <stdio.h>
int map[101][101], visited[101], n;
void dfs(int x) {
    visited[x] = 1;
    for (int i = 1; i <= n; i++)
        if (map[x][i] && !visited[i]) dfs(i);
}
int main() {
    int m, a, b, count = 0;
    scanf("%d %d", &n, &m);
    for (int i = 0; i < m; i++) {
        scanf("%d %d", &a, &b);
        map[a][b] = map[b][a] = 1;
    }
    dfs(1);
    for (int i = 2; i <= n; i++) if (visited[i]) count++;
    printf("%d\n", count);
    return 0;
}